src/handshake/client.rs

```diff
@@ -32,10 +32,10 @@ pub struct Client<'a, T> {
 	socket: T,
 	/// The HTTP host to send the handshake to.
 	host: &'a str,
-	/// The HTTP host ressource.
+	/// The HTTP host resource.
 	resource: &'a str,
 	/// The HTTP headers.
-	headers: &'a [Header<'a>],
+	headers: Option<&'a HeaderMap>,
```

Comment:
it doesn't makes sense to use `httparse::Header` to construct new headers, the type is used for parsed headers and doesn't check whether the header is valid or not.

however, it still possible to enter some conflicting headers such as `host`, `websocket upgrade` and similar as these are configured separately.